### PR TITLE
agent logging : resolves (agent-1404)

### PR DESCRIPTION
**Pull request by NehaNaithani**
configurable log level, acronym defined for events

### CODE REVIEW

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/field_agent/FieldAgent.java [hunk 1]
```diff
@@ -1059,10 +1059,10 @@ public void start() {
             processRoutes(microservices);
         }
 
-        new Thread(pingController, "FieldAgent : Ping").start();
-        new Thread(getChangesList, "FieldAgent : GetChangesList").start();
-        new Thread(postStatus, "FieldAgent : PostStatus").start();
-        new Thread(postDiagnostics, "FieldAgent : PostDiagnostics").start();
+        new Thread(pingController, "FAPC").start();
```
Comment (baghbidi) [inline iofog-agent-daemon/src/main/java/org/eclipse/iofog/field_agent/FieldAgent.java]: Please define these abbreviations as constants in a separate file

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/exception/AgentException.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Edgeworx, Inc.
```
Comment (baghbidi) [inline iofog-agent-daemon/src/main/java/org/eclipse/iofog/exception/AgentException.java]: 2019?

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/field_agent/FieldAgent.java (hunk 1 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/field_agent/FieldAgent.java]: ok

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/exception/AgentException.java (hunk 2 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/exception/AgentException.java]: updated

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/command_line/CommandLineConfigParam.java [hunk 3]
```diff
@@ -43,6 +43,7 @@
     LOG_DISK_CONSUMPTION_LIMIT ("10", "l","log_disk_consumption_limit", "logLimit"),
     LOG_DISK_DIRECTORY ("/var/log/iofog-agent/", "ld","log_disk_directory", "logDirectory"),
     LOG_FILE_COUNT ("10", "lc","log_file_count", "logFileCount"),
+    LOG_LEVEL ("SEVERE", "lv","log_level", "loglevel"),
```
[the remark this answers is not in the document]
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/command_line/CommandLineConfigParam.java]: Help is updated this is outdated file

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/utils/configuration/Configuration.java [hunk 4]
```diff
@@ -987,10 +1033,12 @@ public static String getConfigReport() {
         result.append(buildReportLine(getConfigParamMessage(LOG_DISK_DIRECTORY), logDiskDirectory));
         // log files count
         result.append(buildReportLine(getConfigParamMessage(LOG_FILE_COUNT), format("%d", logFileCount)));
+     // log files count
```
[the remark this answers is not in the document]
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/utils/configuration/Configuration.java]: updated

File: iofog-agent-daemon/src/main/java/org/eclipse/iofog/diagnostics/ImageDownloadManager.java [hunk 5]
```diff
@@ -63,24 +65,27 @@ public static void createImageSnapshot(Orchestrator orchestrator, String microse
...
+                logError(MODULE_NAME, "Unable send image snapshot path",
+                		new AgentSystemException("Unable send image snapshot path", e));
             }
         }
+        LoggingService.logInfo(MODULE_NAME, "Finished Create image snapshot");
     }
 
     private static File getFileByImagePath(String path) {
-
+    	LoggingService.logInfo(MODULE_NAME, "Finished get file by image path");
```
[the remark this answers is not in the document]
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/diagnostics/ImageDownloadManager.java]: Updated

[the remark this answers is not in the document]
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/diagnostics/ImageDownloadManager.java]: Updated

[the remark this answers is not in the document]
Comment (NehaNaithani) [reply iofog-agent-daemon/src/main/java/org/eclipse/iofog/diagnostics/ImageDownloadManager.java]: Updated

**Pull request merged**